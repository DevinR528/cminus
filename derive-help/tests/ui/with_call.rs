fn debug_fail(foo: &usize, bar: &usize) -> &'static str {
    "hello"
}

#[derive(derive_help::Debug)]
enum Enum2<T> {
    A(T),
    #[dbg_with(debug_fail)]
    B(usize, usize),
    #[dbg_ignore]
    Car {
        foo: usize,
        bar: bool,
    },
}

fn debug_pass<T: std::fmt::Debug>(foo: T) -> &'static str {
    "hello"
}

#[derive(derive_help::Debug)]
struct Foob<T> {
    a: bool,
    #[dbg_with(debug_pass)]
    b: T,
    #[dbg_ignore]
    c: T,
}

#[derive(derive_help::Debug)]
struct Foo<T>(#[dbg_with(debug_pass)] T);

#[derive(derive_help::Debug)]
struct FooDouble<T>(#[dbg_with(debug_pass)] T, usize);

fn debug_no_lt<T: std::fmt::Debug>(foo: T) -> String {
    "hello".into()
}

struct Bar;
impl Bar {
    fn print(&self) -> &'static str {
        "bar"
    }
}

#[derive(derive_help::Debug)]
enum EnumNoLt {
    #[dbg_with(Bar::print)]
    A(Bar),
    #[dbg_with(debug_no_lt)]
    B(usize),
}

fn main() {}
