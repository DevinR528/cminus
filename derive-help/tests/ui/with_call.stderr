error: can only call debug_with on one "field" at a time
 --> tests/ui/with_call.rs:8:2
  |
8 | /     #[dbg_with(debug_fail)]
9 | |     B(usize, usize),
  | |___________________^
